### PR TITLE
Stp 885 refactor runner into typescript

### PR DESCRIPTION
**Pull request by bertrandmc**
# Description

Ticket: STP-830 Merge Engine and Runner.

## Changes
- Move engine workspace code to runner's `server/plugins`.
- Merge runner's `Build` plugin with engine.
- Refactor Engine's ComponentCollection to load components based on type.
- Fix issues with circular dependencies to allow `components/index` to export all named components.
- Rename all files that were not following JS/React file naming convention.
- Delete Engine workspaces and any references to it.
- Remove CircleCI configurations related to Engine workspace. 
- Initial Runner refactoring into typescript (I  have converted all files and added loads of types but this is still an ongoing task, I have also to continue looking into the engine merged code to reduce duplication and refactor).

## Type of change

- [X] Technical Debt (non-breaking change)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

- [X] I have added some more tests and adjusted existing ones.
- [X] New and existing unit tests pass locally with my changes.
- [X] I have runned the application locally with a large JSON sample form which was executed without issues. 

# Checklist:

- [X] My changes do not introduce any new linting errors
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation and versioning
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I have rebased onto master and there are no code conflicts

Screenshot 2020-11-11 at 17 20 50 :)

### CODE REVIEW

**Comment by bertrandmc** [comment]
@jenbutongit once approved we'll need your admin power to merge this without the `ci/circleci: engine-build-lint-test`. I have removed it from circle's config but until that get's merged we'll have to waiting status above.

**Comment by jenbutongit** [comment]
@bertrandmc - I've removed that check now

File: runner/src/server/plugins/engine/pages/README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Page Controllers
+
+TODO: Refactor to load page controllers from memory instead of depending on path and dynamic require.
```
Comment (jenbutongit) [inline runner/src/server/plugins/engine/pages/README.md:3]: +1

File: runner/src/server/plugins/engine/pages/home.ts [hunk 2]
```diff
@@ -1,20 +1,12 @@
 import Page from "./page";
+import { HapiRequest, HapiResponseToolkit } from "../../../types";
```
Comment (jenbutongit) [inline runner/src/server/plugins/engine/pages/home.ts:2]: no need to action now, but we should alias/remap so we don't have to do `../../../xxx` 🤓

**Review by jenbutongit — APPROVED** [review]
lgtm!! Thank you!! (sorry summary vm is so huge.. we'll tackle it eventually!)

We should collect/review our //TODO's at some point too.

I don't have time to spin up this branch and check but looks like you've tested with a file anyway

**Comment by bertrandmc** [comment]
Thanks Jen. My next PR will break things down further and add more types and we can keep iterating. In my optionion It is not realistic to try to push everything (refactor and all types) in a single PR, otherwise I'd be working on it for weeks.

**Pull request merged**